js/jquery.mapael.js

```diff
@@ -2057,43 +2061,122 @@
-                    var ratio = currentStep / steps;
-                    self.paper.setViewBox(cx + dx * easingFormula(ratio),
-                        cy + dy * easingFormula(ratio),
-                        cw + dw * easingFormula(ratio),
-                        ch + dh * easingFormula(ratio), false);
-                    if (currentStep++ >= steps) {
-                        clearInterval(self.animationIntervalID);
-                        clearTimeout(self.zoomTO);
-                        self.zoomTO = setTimeout(function () {
-                            self.$map.trigger("afterZoom", {x1: x, y1: y, x2: (x + w), y2: (y + h)});
-                        }, 150);
+
+            var easingFormula = Raphael.easing_formulas[easingFunction || "linear"];
+
+            // To avoid another frame when elapsed time approach end (2%)
+            var durationWithMargin = duration - (duration * 2 / 100);
+
+            var tStart = (new Date()).getTime();
+            var computeNextStep = function () {
+                // Cancel any remaining animationFrame
+                self.cancelAnimationFrame(self.animationIntervalID);
+
+                var elapsed = (new Date()).getTime() - tStart;
+                if (elapsed < durationWithMargin) {
+                    // Compute ratio according to elasped time and easing formula
+                    var ratio = easingFormula(elapsed / duration);
+                    self.setViewBox(
+                        cx + dx * ratio, cy + dy * ratio,
+                        cw + dw * ratio, ch + dh * ratio
+                    );
+                    self.animationIntervalID = self.requestAnimationFrame(computeNextStep);
+                } else {
+                    // Set the viewbox to final state
+                    self.setViewBox(x, y, w, h);
+                    // Trigger afterZoom event
+                    self.$map.trigger("afterZoom", {x1: x, y1: y, x2: (x + w), y2: (y + h)});
+                }
+            };
+
+            self.cancelAnimationFrame(self.animationIntervalID);
+            self.animationIntervalID = self.requestAnimationFrame(computeNextStep);
+        },
+
+        /*
+         * requestAnimationFrame polyfill
+         * Based on https://gist.github.com/jlmakes/47eba84c54bc306186ac1ab2ffd336d4
+         * and also https://gist.github.com/paulirish/1579671
+         *
+         * _requestAnimationFrameFn and _cancelAnimationFrameFn hold the current functions
+         * But requestAnimationFrame and cancelAnimationFrame shall be called since
+         * in order to be in window context
+         */
+        // The requestAnimationFrame polyfill'd function
+        _requestAnimationFrameFn: (function () {
+            var polyfill = (function () {
+                var clock = (new Date()).getTime();
+
+                return function (callback) {
+                    var currentTime = (new Date()).getTime();
+
+                    if (currentTime - clock > 16) {
```

Comment:
I think you should add a comment here in order to not to forgot the purpose of this test

Reply:
Ok, I will add that

Reply:
Thanks !